### ISSUE TITLE
Checking dependencies for second build should happen before making a new tag and accessing git

### ISSUE THREAD

**Issue by flip111** [purescript/spago#1142]
About the log at the bottom of this post

1.
`Pushing tag 'v0.4.3' to the remote` this should happen after dependencies for step `Building again with the build plan from the solver...` are checked. 0.4.3 is now a wasted version (unless i want to do complicated things with my git history to remove the tag).

2.
One could also give the solution which is to run `npx spago publish` before publish. Or during publish the tool could ask to upgrade your package set automatically. Of course you need another commit after that because `spago.yaml` was changed.

For more information in the situation of the log `.spago/packages/arrays-7.3.0/` was not available but `.spago/packages/arrays-7.2.1/` was available which was ok for normal build (not the one "build plan from the solver")

I'm a bit confused why it's building twice since it could already figure out that upgrade of package set is required.

```
» npx spago publish                                                                                                                                                                                                                                                    1 [15:07:47]
Reading Spago workspace configuration...
Read the package set from the registry

✅ Selecting package to build: arrays-extra

Downloading dependencies...
Building...
           Src   Lib   All
Warnings     0     0     0
Errors       0     0     0

✅ Build succeeded.

Pushing tag 'v0.4.3' to the remote
Username for 'https://github.com': flip111
Password for 'https://flip111@github.com':
Building again with the build plan from the solver...
Building...
purs compile: No files found using pattern: .spago/packages/arrays-7.3.0/src/**/*.purs
[1/13 ModuleNotFound] .spago/packages/filterable-5.0.0/src/Data/Compactable.purs:18:1

  18  import Data.Array ((!!))
      ^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[2/13 ModuleNotFound] .spago/packages/filterable-5.0.0/src/Data/Compactable.purs:19:1

  19  import Data.Array.ST as STA
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array.ST was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[3/13 ModuleNotFound] .spago/packages/filterable-5.0.0/src/Data/Compactable.purs:20:1

  20  import Data.Array.ST.Iterator as STAI
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array.ST.Iterator was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[4/13 ModuleNotFound] .spago/packages/filterable-5.0.0/src/Data/Filterable.purs:23:1

  23  import Data.Array (partition, mapMaybe, filter) as Array
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[5/13 ModuleNotFound] .spago/packages/ordered-collections-3.1.1/src/Data/Set/NonEmpty.purs:29:1

  29  import Data.Array.NonEmpty (NonEmptyArray)
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array.NonEmpty was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[6/13 ModuleNotFound] src/Data/Array/Extra.purs:6:1

  6  import Data.Array (all, deleteBy, elem, filter, foldl, foldr, intersectBy, length, null, snoc, sortBy, unionBy)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[7/13 ModuleNotFound] src/Data/Array/Extra.purs:8:1

  8  import Data.Array.NonEmpty (NonEmptyArray)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array.NonEmpty was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[8/13 ModuleNotFound] src/Data/Array/Extra.purs:9:1

  9  import Data.Array.NonEmpty as NEA
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array.NonEmpty was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[9/13 ModuleNotFound] src/Data/Array/Extra/All.purs:3:1

  3  import Data.Array (filter, length, range, zip, foldl)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[10/13 ModuleNotFound] src/Data/Array/Extra/First.purs:5:1

  5  import Data.Array (findIndex, unsafeIndex, snoc, cons, take, insertAt, drop, uncons)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[11/13 ModuleNotFound] src/Data/Array/Extra/Insert.purs:5:1

  5  import Data.Array (cons, elem, findIndex, insertBy, snoc, length)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[12/13 ModuleNotFound] src/Data/Array/Extra/Last.purs:5:1

  5  import Data.Array (findLastIndex, unsafeIndex, snoc, cons, take, drop, insertAt, unsnoc)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

[13/13 ModuleNotFound] src/Data/Array/Extra/Unsafe.purs:12:1

  12  import Data.Array (head)
      ^^^^^^^^^^^^^^^^^^^^^^^^

  Module Data.Array was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.

           Src   Lib   All
Warnings     0     0     0
Errors       8     5    13

❌ Failed to build.
```

**Comment by f-f** [comment]
Are you on the latest version? I believe we fixed this in #1121

**Comment by f-f** [comment]
Ah sorry, I misread - I believe this is a duplicate of #1110

**Comment by flip111** [comment]
Ok i opened that issue for another reason. But seemed that your proposed solution `We should be fetching them before issuing a rebuild` would solve this one too
